buildscripts/indexsymbols.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python3
+#########################################################################################
+#
+# Description:  This contains helper methods for source indexing
+#
+# Requirement:
+#               python 3.x
+#               srctool.exe and pdbstr.exe
+#
+#############################################################################################
+
+import os.path
+import argparse
+import subprocess
+from subprocess import Popen, PIPE
+
+def write_index(index_filename, tag_version):
+    """This writes to a temporary index file for later use
+
+    For example
+
+    SRCSRV: ini ------------------------------------------------
+    VERSION=1
+    SRCSRV: variables ------------------------------------------
+    PATH=%var2%
+    SRCSRVTRG=%TARG%\%PDBVERSION%\%fnbksl%(%var2%)
+    SRCURL=https://raw.githubusercontent.com/Microsoft/msphpsql/%SRCVERSION%/source/%PATH%
+    SRCSRVCMD=powershell -Command "$r=New-Object -ComObject Msxml2.XMLHTTP; $r.open('GET', '%SRCURL%', $false); $r.send(); [io.file]::WriteAllBytes('%SRCSRVTRG%', $r.responseBody)"

```

Comment:
What does this command do? Is this the command that actually does the source indexing?

Reply:
This inserts the powershell command into individual source file in the pdb file such that when that source file is needed, it will be downloaded from the right place based on the url